### PR TITLE
Support apollo grouped dynamic configurations

### PR DESCRIPTION
**Pull request by wankai123**
### Support apollo grouped dynamic configurations
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [X] Update the documentation to include this new feature.
- [X] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

see: `dynamic-config-apollo.md`

### CODE REVIEW

**Review by wu-sheng — DISMISSED** [review]

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+public class ITApolloConfigurationTest {
+    private final Yaml yaml = new Yaml();
+    private final String token = "f71f002a4ff9845639ef655ee7019759e31449de";
+    private final CloseableHttpClient httpClient = HttpClients.createDefault();
+    private final ResponseHandler responseHandler = new BasicResponseHandler();
+
+    private String baseUrl;
+    private ApolloConfigurationTestProvider provider;
+
+    @ClassRule
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:63]: Do not use ClassRule, use Rule instead

Comment (wankai123) [reply oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:63]: why? deploy this cost much time, I think the 2 tests can share the container env.

Comment (kezhenxu94) [reply oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:63]: They can share the same container for sure, the same as other tests like etcd, nacos, etc., but the fact is that they were flaky with a various of reasons before

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    private final Yaml yaml = new Yaml();
+    private final String token = "f71f002a4ff9845639ef655ee7019759e31449de";
+    private final CloseableHttpClient httpClient = HttpClients.createDefault();
+    private final ResponseHandler responseHandler = new BasicResponseHandler();
+
+    private String baseUrl;
+    private ApolloConfigurationTestProvider provider;
+
+    @Rule
+    public final DockerComposeContainer environment =
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:66]:
```suggestion
    public final DockerComposeContainer<?> environment =
```

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        new DockerComposeContainer(new File(ITApolloConfigurationTest.class
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:67]:
```suggestion
        new DockerComposeContainer<>(new File(ITApolloConfigurationTest.class
```

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+            final StringEntity entity = new StringEntity("{\n" + "    \"key\":\"test-module.default.testKey\",\n" + "    \"value\":\"3000\",\n" + "    \"comment\":\"test key\",\n" + "    \"dataChangeCreatedBy\":\"apollo\"\n" + "}");
+            createConfigPost.setEntity(entity);
+            String createResponse = null;
+            //retry to wait apollo adminserver registered
+            for (int r = 1; r <= 10 && createResponse == null; r++) {
+                TimeUnit.SECONDS.sleep(5);
+                log.info("try createItem, times...: {}", r);
+                createResponse = this.httpExec(createConfigPost, responseHandler);
+                log.info("createResponse: {}", createResponse);
+
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:120]:
```suggestion
```

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        } catch (IOException | InterruptedException e) {
+            log.error(e.getMessage(), e);
+            fail(e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("StatementWithEmptyBody")
+    @Test(timeout = 100000)
+    public void shouldReadUpdated4Group() {
+
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:155]:
```suggestion
```

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+            createConfigPost.setEntity(entityItem1);
+
+            String createResponseItem1 = null;
+            //retry to wait apollo adminserver registered
+            for (int r = 1; r <= 10 && createResponseItem1 == null; r++) {
+                TimeUnit.SECONDS.sleep(5);
+                log.info("try createItem, times...: {}", r);
+                createResponseItem1 = this.httpExec(createConfigPost, responseHandler);
+                log.info("createResponse: {}", createResponseItem1);
+
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:172]:
```suggestion
```

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+            deleteConfigRequest.setHeader("Content-Type", "application/json;charset=UTF-8");
+            httpClient.execute(deleteConfigRequest);
+
+            final String releaseDeleteResponse = (String) httpClient.execute(releaseConfigRequest, responseHandler);
+            log.info("releaseDeleteResponse: {}", releaseDeleteResponse);
+            for (String v = provider.groupWatcher.groupItems().get("item1"); v != null; v = provider.groupWatcher.groupItems().get("item1")) {
+            }
+            assertNull(provider.groupWatcher.groupItems().get("item1"));
+            assertEquals("200", provider.groupWatcher.groupItems().get("item2"));
+
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:206]:
```suggestion
```

File: oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+                }
+            });
+        }
+    }
+
+    //for retry
+    private String httpExec(HttpUriRequest request, ResponseHandler responseHandler) {
+        try {
+            return (String) this.httpClient.execute(request, responseHandler);
+        } catch (IOException e) {
```
Comment (kezhenxu94) [inline oap-server/server-configuration/configuration-apollo/src/test/java/org/apache/skywalking/oap/server/configuration/apollo/ITApolloConfigurationTest.java:243]:
```suggestion
        } catch (IOException ignored) {
```

**Review by kezhenxu94 — COMMENTED** [review]
Some nits

**Comment by wankai123** [comment]
due to start the container cost a long time, if `@ClassRule` is unstable or can't save enough time, I think we have to ignore apollo's IT cases. @wu-sheng @kezhenxu94

**Comment by wu-sheng** [comment]
How much time are you talking about for CI? Could you compare between the previous time and current?

**Comment by wankai123** [comment]
> How much time are you talking about for CI? Could you compare between the previous time and current?

start the container needs more than 1 minute, if don't share the container, the 2 cases could cost more than 3 minutes

**Comment by wu-sheng** [comment]
I am OK to remove it from CI process, but we should keep it runnable locally after minimal change when we need it.

**Comment by wankai123** [comment]
CI Passed, it's cost 2-4min

**Comment by wu-sheng** [comment]
@kezhenxu94 What do you think? It is already 25 mins for CI with IT.

**Comment by kezhenxu94** [comment]
> @kezhenxu94 What do you think? It is already 25 mins for CI with IT.

It's ok for me if it's stable enough. I can find other means to reduce CI waiting time, but flaky tests are intolerable to me. It's ok to me now. Let's see how it will behave.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**